llvm/lib/Target/X86/X86FrameLowering.cpp

```diff
@@ -1818,18 +1852,30 @@ void X86FrameLowering::emitPrologue(MachineFunction &MF,
   // Skip the callee-saved push instructions.
   bool PushedRegs = false;
   int StackOffset = 2 * stackGrowth;
+  MachineBasicBlock::iterator LastCSPush = MBBI;
+  auto IsCSPush = [&](const MachineBasicBlock::iterator &MBBI) {
+    if (MBBI == MBB.end() || !MBBI->getFlag(MachineInstr::FrameSetup))
+      return false;
+    unsigned Opc = MBBI->getOpcode();
+    return Opc == X86::PUSH32r || Opc == X86::PUSH64r || Opc == X86::PUSHP64r ||
+           Opc == X86::PUSH2 || Opc == X86::PUSH2P;
+  };
 
-  while (MBBI != MBB.end() && MBBI->getFlag(MachineInstr::FrameSetup) &&
-         (MBBI->getOpcode() == X86::PUSH32r ||
-          MBBI->getOpcode() == X86::PUSH64r)) {
+  while (IsCSPush(MBBI)) {
     PushedRegs = true;
     Register Reg = MBBI->getOperand(0).getReg();
+    LastCSPush = MBBI;
```

Comment:
You don't need to save the last MBBI. All you need it just it's opcode and register.

Reply:
I think no difference. We need its opcode and needs its register, why not save the last MBBI directly?

Reply:
We use `MBBI` in a lot of place to operate MI instructions. It gives me the impression you will do similar thing here.

Reply:
Okay. I can change the `MachineBasicBlock::iterator` to `MachineBasicBlock::const_iterator` to avoid misusing.

Reply:
Done